src/realm/object-store/schema.cpp

```diff
@@ -302,12 +302,29 @@ void Schema::zip_matching(T&& a, U&& b, Func&& func) noexcept
             ++j;
         }
     }
+
     for (; i < a.size(); ++i)
         func(&a[i], nullptr);
     for (; j < b.size(); ++j)
         func(nullptr, &b[j]);
 }
 
+void Schema::append_missing_objects(const Schema& other)
+{
+    size_t i = 0;
+    Schema missing_objects;
+    Schema& current_schema = *this;
+    size_t size = std::min(current_schema.size(), other.size());
+
+    for (; i < size; ++i)
+        if (current_schema[i].name.compare(other[i].name) > 0)
```

Comment:
They're in sorted order, but this comparison still doesn't work. Appending them to the back is also wrong because they're sorted.

Reply:
OK, so this is a problem, I thought the order of the non-matching objects did not matter. Even appending them sorted is not going to work. 
It does not seem we allow schemas to be iterated over, so I guess we either merge things respecting the order or I can add a separate vector of non-matching object schemas in order to extend the search in case the object name does not match.

Reply:
I am appending the new objects into another vector of unmatched schema objects, and I have changed how we search them. The downside of this approach is that, the unmatching objects won't be accessible to the SDK, but knowing the class name would suffice, which I believe is enough to implement the functionality required.

Reply:
This loop still doesn't work. Suppose `current_schema` is `{"A", "C", "D"}` and `other` is `{"A", "B", "C", "D"}`. At the end of this function, `m_others` will be `{"B", "C", "D"}`.